libs/shared-utils/src/Application/YoutubeService.ts

```diff
@@ -0,0 +1,31 @@
+import axios, { AxiosResponse } from "axios";
+import { YoutubeException } from "./Exception/GoogleWorkSpaceException";
+
+export const getYoutubeEmbedUrl = async (url: URL): Promise<string> => {
+    const link = url.toString();
+    if (isEmbedableYoutubeLink(url)) return link;
+    return await axios
+        .get(`https://www.youtube.com/oembed?url=${link}&format:json`)
+        .then((res: AxiosResponse<{ html: string | undefined }>) => {
+            const html = res.data.html;
+            if (html == undefined) throw new Error("No html found");
+            const div = document.createElement("div");
+            div.insertAdjacentHTML("beforeend", html);
+            const iframe: HTMLIFrameElement = div.firstChild as HTMLIFrameElement;
+            return iframe.src;
+        })
```

Comment:
This piece of code is generic to the oEmbed protocol and not specific to Youtube. We could maybe put it in a more generic service.

Reply:
if we integrate new service with `oembed` protocol, we could create a generic service and use it to our applications